.travis.yml

```diff
@@ -3,13 +3,19 @@ compiler:
   - gcc
 before_script:
   - git submodule update --init --recursive
-  - mkdir -p build
-  - cd build
-  - cmake .. -DCMAKE_BUILD_TYPE=Debug
+  - mkdir -p build/debug
+  - mkdir -p build/release
+  - cd build/debug
+  - cmake ../.. -DCMAKE_BUILD_TYPE=Debug
+  - cd ../release
+  - cmake ../.. -DCMAKE_BUILD_TYPE=Release
 script:
   - cmake --build .
   - ctest -V
```

Comment:
Maybe build and test both `release` and `debug` (or `DebugWithCoverage`) in `script`, not `after_success`.  If the `debug` build or tests fail, that's not a a successful build.  `after_success` should just do the upload of the coverage outputs.

Reply:
I wasn't sure what the usual etiquette is. But, I agree with this.

The build-type will now be `DebugWithCoverage`, and just `Debug` is left alone. Good call.